src/pages/HomePage.js

```diff
@@ -1,21 +1,58 @@
-import React, { Component } from "react";
-
-import Banner from "../components/HomePage/Banner";
-import ButtonBar from "../components/HomePage/ButtonBar";
-import SleepyActionOutput from "../components/HomePage/SleepyActionOutput";
-import RandomDogOutput from "../components/HomePage/RandomDogOutput";
-
-class HomePage extends Component {
-    render() {
-        return (
-            <React.Fragment>
-                <Banner />
-                <ButtonBar />
-                <SleepyActionOutput />
-                <RandomDogOutput />
-            </React.Fragment>
-        );
-    }
-}
+import React from "react";
+
+import MainView from "../components/HomePage/MainView";
+import SearchResults from "../components/HomePage/SearchResultsArea/SearchResults";
+import ProductDescription from "../components/HomePage/ProductPlacementArea/ProductDescription";
+
+export const scrollToProductDescription = (productDescriptionRef) => {
+    productDescriptionRef.current.scrollIntoView({
+        behavior: "smooth",
+        block: "start",
+    });
+};
+export const scrollToSearchResults = (searchResultsRef) => {
+    searchResultsRef.current.scrollIntoView({
+        behavior: "smooth",
+        block: "start",
+    });
+};
+
+const HomePage = () => {
+
+    const [productDescriptionRef, setProductDescriptionRef] = React.useState(null);
+    const [searchResultsRef, setSearchResultsRef] = React.useState(null);
+
+    // Setting the offers, probably needs re-thinking
+    // Maybe use React.useEffect for offer loading
```

Comment:
I did some refactors regarding the `Offers`. The offers will be fetched once the user searches (might change later, if we want to show some preferred ones from the beginning), but the point is: they will be stored in the redux state, and this component won't be the one responsible for fetching them, the `SearchArea` will do it `onSubmit`. This component will connect to the store and pass the current offers to the `SearchResults`. If the offers array is empty (initial state), the `SearchResults` renders the skeleton version, to make it seem they are loading. However, this is not enough, because it does not make sense to have x offers loading without having made a search yet. So, the `SearchResults` has now a `show` flag, which is enabled when the user searches for something, triggering the render of the `SearchResults`component and, while the offers are not ready yet, there will be a skeleton, when they arrive, they will be shown.

Reply:
Makes sense!